graph/builder.go

```diff
@@ -675,45 +675,20 @@ func (b *Builder) handleNetworkUpdate(vb *ValidationBarrier,
 	defer b.wg.Done()
 	defer vb.CompleteJob()
 
-	// If this message has an existing dependency, then we'll wait until
-	// that has been fully validated before we proceed.
-	err := vb.WaitForDependants(update.msg)
-	if err != nil {
-		switch {
-		case IsError(err, ErrVBarrierShuttingDown):
-			update.err <- err
-
-		case IsError(err, ErrParentValidationFailed):
-			update.err <- NewErrf(ErrIgnored, err.Error()) //nolint
-
-		default:
-			log.Warnf("unexpected error during validation "+
-				"barrier shutdown: %v", err)
-			update.err <- err
-		}
-
-		return
-	}
```

Comment:
just want to note that in future, we will likely have the Builder's AddNode/AddChannel/UpdateChannel calls be exposed via RPC for clients who dont want to sync the graph themselves. So initially i was thinking we may want to keep the validation barrier for that use case but after having thought about it more, i think it is fine to remove since those clients will always be trusted AND those calls will be done in sync with the gossiper calls performed by those clients. 

So, ACK from my side for removal here 👍

Reply:
As long as they're done in sync with the gossiper calls and the semaphore logic is used, that all sounds good and should work fine.